src/main/java/com/google/api/codegen/util/CommentReformatting.java

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import com.google.api.codegen.CommentPatterns;
+import com.google.common.base.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** A collection of static helper methods for reformatting comments. */
+public class CommentReformatting {
+
+  public static String CLOUD_URL_PREFIX = "https://cloud.google.com";
+
+  private CommentReformatting() {}
+
+  public static String reformatAbsoluteMarkdownLinks(String comment, String linkFormat) {
+    return reformatPattern(
+        comment, CommentPatterns.ABSOLUTE_LINK_PATTERN, reformatLinkFunction(linkFormat, ""));
+  }
+
+  public static String reformatCloudMarkdownLinks(String comment, String linkFormat) {
+    return reformatPattern(
+        comment,
+        CommentPatterns.CLOUD_LINK_PATTERN,
+        reformatLinkFunction(linkFormat, CLOUD_URL_PREFIX));
+  }
+
+  public static String reformatPattern(
+      String comment, Pattern pattern, Function<Matcher, String> replacementFunction) {
+    StringBuffer sb = new StringBuffer();
+    Matcher m = pattern.matcher(comment);
+    if (!m.find()) {
+      return comment;
+    }
+    do {
+      m.appendReplacement(sb, replacementFunction.apply(m));
+    } while (m.find());
+    m.appendTail(sb);
+    return sb.toString();
+  }
+
+  public static Formatter of(String comment) {
+    return new Formatter(comment);
+  }
+
+  /**
+   * The Formatter object allows the static methods in CommentReformatting to be used via a fluent
+   * interface.
```

Comment:
I would prefer for this class to be opinionated and only provide the fluent interface, instead of both static methods + the fluent interface. Thus, I would suggest lifting Formatter implementation up and making the static methods private. Maybe rename the existing `CommentFormatter` interface to `LanguageCommentFormatter`, so that the name `CommentFormatter` is free.